### ISSUE TITLE
Concerning the explicit result of "Close"

### ISSUE THREAD

**Issue by aplteam** [the-carlisle-group/Acre-Desktop#185]
According to the documentation "Close" is supposed to  return the number of projects still tracked.

Firstly, this is not true when the command failes, for example due to an invalid path: in that case `''` is returned.

Secondly, the documentation does not mention what happens when more than one project is specified: only the first one is closed, the other(s) are ignored.

I wonder whether it would be better to return a success flag: 1 for "successfully closed" and 0 for failure.

Finally I would be happy to have the options to close all project, no matter what, for example with an `-all` flag.

**Comment by PhilLast** [comment]
**appropriateness of the result**

I can only assume you are not looking at the log as the command will issue the messages, e.g.
```
2019-08-06 08:06:16 Project not tracked
2019-08-06 08:06:16 ]Close #.arthur
```
The acre framework is working out to which project to hand this _project_ method. Not finding one it issues the message and returns empty. This is what it's always done for _project_ methods.

However I've just changed all of the API functions that mirror either _item_ or _project_ commands to signal, e.g.
```
      ⎕SE.acreAPI.changes '#.arthur.dent'
No appropriate items supplied
      ⎕SE.acreAPI.changes'#.arthur.dent'
      ∧
      ⎕SE.acreAPI.close '#.arthur'
Project not tracked
      ⎕SE.acreAPI.close'#.arthur'
      ∧
```
The lighter framework requires each function to handle this individually so this _could_ mimic the command by issuing a message but differ by returning an appropriate "zero" result - i.e. one that is unlikely to cause LENGTH, RANK or DOMAIN errors in the caller. In the above `close` would continue to return the number left open while `changes` returned `0 1⍴⊂''`

**success of the request**

You can be assured that a project matching the argument wasn't or is no longer open so in that sense it is always a success. A zero currently indicates that no more are open.

**closure of _-all_**

Good idea, why not! The current way I've translated command options to the API is via an optional left argument that tends to be a positional boolean list. For `close` this would convert as an optional `1` for `-all`. OK?

**Comment by aplteam** [comment]
> I can only assume you are not looking at the log as the command will issue the messages, e.g.

Indeed. I run into this not by issuing the commands in the session but by working on the test cases.
If I can help it I don't want to analyze the log. Signalling an error in case the project specified does not exist is fine in this context. I would like to have a similar mechanism for the user command as such.

>  For close this would convert as an optional 1 for -all. OK?
Yes!

**Comment by PhilLast** [comment]
Acre is parameterised as far as possible as I've so far been able to make it. All user commands are handled by one framework and all API functions by another. As much commonality is bundled into those frameworks as possible but they are further categorised as _shared_, _project_ and _item_ methods each of which also bears some burden of functionality.

Unless I start instituting special cases for different commands/functions in different circumstances 
the ten _project_/_item_ methods: `Close`, `Refresh` & `Stats`and `Changes`, `Erase`, `PurgeHistory`, `Redo`, `Restore`, `SetChanged` & `Undo` will have to behave similarly in this regard.

> **That is they will all continue to message and return _something_ or they will all signal an error.**

**Comment by PhilLast** [comment]
Adding `-all` to `]acre.Close` and `acreAPI.close` can most easily (possibly, _only_) be achieved through the framework(s). This would mean that `]acre.Refresh`, `acreAPI.refresh`, `]acre.Stats` and `acreAPI.stats` would also share the benefit.

I'm ashamed to say that @norberturkiewicz requested `]acre.Stats all` possibly even before we ever included `-options` in acre but I prevaricated and then forgot. I have the uncomfortable feeling that the same request might well have been made regarding `]acre.refresh`.

**Comment by PhilLast** [comment]
```
]acre.stats -all
]acre.close -all
]acre.refresh -all
]acre.config -all
```
all implemented in v6.0.0
